app/main.py

```diff
@@ -1,8 +1,40 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int):
+        self.name = name
+        self.age = age
+
+        Person.people[name] = self
+
+    def set_spouse(self, spouse_name):
+
+        if spouse_name is None:
+            return
+
+        spouse = Person.people.get(spouse_name)
```

Comment:
Consider handling the case where the spouse is not found in the `Person.people` dictionary. Currently, if `spouse_name` is not in the dictionary, `spouse` will be `None`, and the function will silently do nothing. It might be better to raise an exception or log a warning in this case.